CONTRIBUTING.md

```diff
@@ -26,8 +26,11 @@ pdm install
 `pdm venv activate`
 # which will active the venv where you can further work, or prefix the rest of commands with `pdm run`, eg. `pdm run make test`
 
-# run all tests
-make test
+# install dependencies
+make install-deps && make install-deps-test
```

Comment:
Did you have troubles with your venv?
I'm asking because `pdm install` installs everything from the lock. We have dev deps in the lock, so everything should be installed via that command.

Reply:
yes i tried to follow the steps in a fresh directory and things were broken until i explicitly installed the deps.

i can try again, or maybe you can.  iirc i created a new empty dir, cloned the repo into it, and then tried to execute these steps.

I will also say in general i found the relationship between venv and pdm a bit confusing(mostly probably my lack of knowledge, but it may be something other people trip over also).  I found it easiest to go ahead and run `. .venv/bin/activate` and then run the raw `make` commands rather than prefixing everything with `pdm`.  But that may be a personal preference more than anything.

Reply:
in particular iirc langchain-community was reported as missing

Reply:
well it seems to have worked this time so now sure if i screwed something up last time or what.  We can close this for now and revisit it if others report issues following these steps

Reply:
> in particular iirc langchain-community was reported as missing

it is not connected to dev deps - this is weird

> . .venv/bin/activate

as stated in docs, you can do `` `pdm venv activate` `` (with backticks) to activate the env, its the same you did, but when doing it via pdm, you are sure you are activating the right one (in case there are multiple)

It probably obvious, but you can also set an alias to not be bothered with it at all - eg. I'm using ``alias wol='cd ~/projects/lightspeed-service && `pdm venv activate` && code .'``

Reply:
> as stated in docs, you can do `pdm venv activate` (with backticks) to activate the env,

yeah, i had done that initially.. i'm not sure what was going on yesterday.  with a clean dir this morning the steps in the readme seemed to work ok.  sorry for the noise.